### ISSUE TITLE
Error on deeptables import on colab

### ISSUE THREAD

**Issue by jabir-ssh** [DataCanvasIO/DeepTables#82]
Hi! I'm trying to run deeptables on google colab and I'm getting an error related to featuretools.primitives. Here's the full error traceback:

`
/usr/local/lib/python3.8/dist-packages/deeptables/models/__init__.py in <module>
      4 from deeptables.models.config import ModelConfig
      5 from deeptables.models.modelset import ModelInfo, ModelSet
----> 6 from deeptables.models.deeptable import DeepTable
      7 from deeptables.models.deepmodel import DeepModel
      8 from deeptables.models.metainfo import ContinuousColumn, CategoricalColumn

/usr/local/lib/python3.8/dist-packages/deeptables/models/deeptable.py in <module>
     15 from tensorflow.keras.layers import Concatenate, BatchNormalization
     16 
---> 17 from hypernets.tabular import get_tool_box, is_dask_installed
     18 from . import modelset, deepnets
     19 from .config import ModelConfig

/usr/local/lib/python3.8/dist-packages/hypernets/tabular/__init__.py in <module>
      4 """
      5 from ._base import get_tool_box, register_toolbox, register_transformer, tb_transformer
----> 6 from .toolbox import ToolBox
      7 
      8 register_toolbox(ToolBox)

/usr/local/lib/python3.8/dist-packages/hypernets/tabular/toolbox.py in <module>
     24 from . import ensemble as ensemble_
     25 from . import estimator_detector as estimator_detector_
---> 26 from . import feature_generators as feature_generators_
     27 from . import metrics as metrics_
     28 from . import pseudo_labeling as pseudo_labeling_

/usr/local/lib/python3.8/dist-packages/hypernets/tabular/feature_generators/__init__.py in <module>
      3 
      4 """
----> 5 from ._primitives import CrossCategorical, GeoHashPrimitive, DaskCompatibleHaversine, TfidfPrimitive
      6 from ._transformers import FeatureGenerationTransformer

/usr/local/lib/python3.8/dist-packages/hypernets/tabular/feature_generators/_primitives.py in <module>
     15 
     16 
---> 17 class DaskCompatibleTransformPrimitive(primitives.TransformPrimitive):
     18     compatibility = [primitives.Library.PANDAS, primitives.Library.DASK]
     19     return_dtype = 'object'

/usr/local/lib/python3.8/dist-packages/hypernets/tabular/feature_generators/_primitives.py in DaskCompatibleTransformPrimitive()
     16 
     17 class DaskCompatibleTransformPrimitive(primitives.TransformPrimitive):
---> 18     compatibility = [primitives.Library.PANDAS, primitives.Library.DASK]
     19     return_dtype = 'object'
     20     commutative = True

AttributeError: module 'featuretools.primitives' has no attribute 'Library'
`

I'd be grateful for your help with this.

**Comment by oaksharks** [comment]
Hi jabir-ssh, you can try another version of the `featuretools` with command:
```
pip install featuretools==1.8.0
```

if it doesn't work, please using `pip list` list your python environment for further debugging.

**Comment by jabir-ssh** [comment]
Thank you @oaksharks for your quick response!
Unfortunately it didn't work and gave the same result. Here's the output of `pip list`:

Package                       Version
----------------------------- ----------------------
absl-py                       1.4.0
aeppl                         0.0.33
aesara                        2.7.9
aiohttp                       3.8.3
aiosignal                     1.3.1
alabaster                     0.7.13
albumentations                1.2.1
altair                        4.2.2
appdirs                       1.4.4
arviz                         0.12.1
astor                         0.8.1
astropy                       4.3.1
astunparse                    1.6.3
async-timeout                 4.0.2
atari-py                      0.2.9
atomicwrites                  1.4.1
attrs                         22.2.0
audioread                     3.0.0
autograd                      1.5
Babel                         2.11.0
backcall                      0.2.0
bcrypt                        4.0.1
beautifulsoup4                4.6.3
bleach                        6.0.0
blis                          0.7.9
bokeh                         2.3.3
branca                        0.6.0
bs4                           0.0.1
CacheControl                  0.12.11
cachetools                    5.3.0
catalogue                     2.0.8
category-encoders             2.6.0
certifi                       2022.12.7
cffi                          1.15.1
cftime                        1.6.2
chardet                       4.0.0
charset-normalizer            2.1.1
click                         7.1.2
clikit                        0.6.2
cloudpickle                   2.2.1
cmake                         3.22.6
cmdstanpy                     1.1.0
colorcet                      3.0.1
colorlover                    0.3.0
community                     1.0.0b1
confection                    0.0.4
cons                          0.4.5
contextlib2                   0.5.5
convertdate                   2.4.0
crashtest                     0.3.1
crcmod                        1.7
cryptography                  39.0.1
cufflinks                     0.17.3
cvxopt                        1.3.0
cvxpy                         1.2.3
cycler                        0.11.0
cymem                         2.0.7
Cython                        0.29.33
daft                          0.0.4
dask                          2022.2.1
datascience                   0.17.5
db-dtypes                     1.0.5
dbus-python                   1.2.16
debugpy                       1.0.0
decorator                     4.4.2
deeptables                    0.2.3.1
defusedxml                    0.7.1
descartes                     1.1.0
dill                          0.3.6
distributed                   2022.2.1
dlib                          19.24.0
dm-tree                       0.1.8
dnspython                     2.3.0
docutils                      0.16
dopamine-rl                   1.0.5
earthengine-api               0.1.340
easydict                      1.10
ecos                          2.0.12
editdistance                  0.5.3
eli5                          0.13.0
en-core-web-sm                3.4.1
entrypoints                   0.4
ephem                         4.1.4
et-xmlfile                    1.1.0
etils                         1.0.0
etuples                       0.3.8
fa2                           0.3.5
fastai                        2.7.10
fastcore                      1.5.28
fastdownload                  0.0.7
fastdtw                       0.3.4
fastjsonschema                2.16.2
fastprogress                  1.0.3
fastrlock                     0.8.1
feather-format                0.4.1
featuretools                  1.8.0
filelock                      3.9.0
firebase-admin                5.3.0
fix-yahoo-finance             0.0.22
Flask                         1.1.4
flatbuffers                   23.1.21
folium                        0.12.1.post1
frozenlist                    1.3.3
fsspec                        2023.1.0
future                        0.16.0
gast                          0.4.0
GDAL                          3.3.2
gdown                         4.4.0
gensim                        3.6.0
geographiclib                 1.52
geopy                         1.17.0
gin-config                    0.5.0
glob2                         0.7
google                        2.0.3
google-api-core               2.11.0
google-api-python-client      2.70.0
google-auth                   2.16.0
google-auth-httplib2          0.1.0
google-auth-oauthlib          0.4.6
google-cloud-bigquery         3.4.2
google-cloud-bigquery-storage 2.18.1
google-cloud-core             2.3.2
google-cloud-datastore        2.11.1
google-cloud-firestore        2.7.3
google-cloud-language         2.6.1
google-cloud-storage          2.7.0
google-cloud-translate        3.8.4
google-colab                  1.0.0
google-crc32c                 1.5.0
google-pasta                  0.2.0
google-resumable-media        2.4.1
googleapis-common-protos      1.58.0
googledrivedownloader         0.4
graphviz                      0.10.1
greenlet                      2.0.2
grpcio                        1.51.1
grpcio-status                 1.48.2
gspread                       3.4.2
gspread-dataframe             3.0.8
gym                           0.25.2
gym-notices                   0.0.8
h5py                          2.10.0
HeapDict                      1.0.1
hijri-converter               2.2.4
holidays                      0.19
holoviews                     1.14.9
html5lib                      1.0.1
httpimport                    0.5.18
httplib2                      0.17.4
httpstan                      4.6.1
humanize                      0.5.1
hypernets                     0.2.5.1
hyperopt                      0.1.2
idna                          2.10
imageio                       2.9.0
imagesize                     1.4.1
imbalanced-learn              0.8.1
imblearn                      0.0
imgaug                        0.4.0
importlib-metadata            6.0.0
importlib-resources           5.10.2
imutils                       0.5.4
inflect                       2.1.0
intel-openmp                  2023.0.0
intervaltree                  2.1.0
ipykernel                     5.3.4
ipython                       7.9.0
ipython-genutils              0.2.0
ipython-sql                   0.3.9
ipywidgets                    7.7.1
itsdangerous                  1.1.0
jax                           0.3.25
jaxlib                        0.3.25+cuda11.cudnn805
jedi                          0.18.2
jieba                         0.42.1
Jinja2                        3.1.2
joblib                        1.2.0
jpeg4py                       0.1.4
jsonschema                    4.3.3
jupyter-client                6.1.12
jupyter-console               6.1.0
jupyter_core                  5.2.0
jupyterlab-widgets            3.0.5
kaggle                        1.5.12
kapre                         0.3.7
keras                         2.11.0
keras-vis                     0.4.1
kiwisolver                    1.4.4
korean-lunar-calendar         0.3.1
langcodes                     3.3.0
libclang                      15.0.6.1
librosa                       0.8.1
lightgbm                      2.2.3
llvmlite                      0.39.1
lmdb                          0.99
locket                        1.0.0
logical-unification           0.4.5
LunarCalendar                 0.0.9
lxml                          4.9.2
Markdown                      3.4.1
MarkupSafe                    2.0.1
marshmallow                   3.19.0
matplotlib                    3.2.2
matplotlib-venn               0.11.7
miniKanren                    1.0.3
missingno                     0.5.1
mistune                       0.8.4
mizani                        0.7.3
mkl                           2019.0
mlxtend                       0.14.0
more-itertools                9.0.0
moviepy                       0.2.3.5
mpmath                        1.2.1
msgpack                       1.0.4
multidict                     6.0.4
multipledispatch              0.6.0
multitasking                  0.0.11
murmurhash                    1.0.9
music21                       5.5.0
natsort                       5.5.0
nbconvert                     5.6.1
nbformat                      5.7.3
netCDF4                       1.6.2
networkx                      3.0
nibabel                       3.0.2
nltk                          3.7
notebook                      5.7.16
numba                         0.56.4
numexpr                       2.8.4
numpy                         1.21.6
oauth2client                  4.1.3
oauthlib                      3.2.2
okgrade                       0.4.3
opencv-contrib-python         4.6.0.66
opencv-python                 4.6.0.66
opencv-python-headless        4.7.0.68
openpyxl                      3.0.10
opt-einsum                    3.3.0
osqp                          0.6.2.post0
packaging                     23.0
palettable                    3.3.0
pandas                        1.5.3
pandas-datareader             0.9.0
pandas-gbq                    0.17.9
pandas-profiling              1.4.1
pandocfilters                 1.5.0
panel                         0.12.1
param                         1.12.3
paramiko                      3.0.0
parso                         0.8.3
partd                         1.3.0
pastel                        0.2.1
pathlib                       1.0.1
pathy                         0.10.1
patsy                         0.5.3
pep517                        0.13.0
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        7.1.2
pip                           22.0.4
pip-tools                     6.6.2
platformdirs                  3.0.0
plotly                        5.5.0
plotnine                      0.8.0
pluggy                        0.7.1
pooch                         1.6.0
portpicker                    1.3.9
prefetch-generator            1.0.3
preshed                       3.0.8
prettytable                   3.6.0
progressbar2                  3.38.0
prometheus-client             0.16.0
promise                       2.3
prompt-toolkit                2.0.10
prophet                       1.1.2
proto-plus                    1.22.2
protobuf                      3.19.6
psutil                        5.9.4
psycopg2                      2.9.5
ptyprocess                    0.7.0
py                            1.11.0
pyarrow                       9.0.0
pyasn1                        0.4.8
pyasn1-modules                0.2.8
pycocotools                   2.0.6
pycparser                     2.21
pyct                          0.5.0
pydantic                      1.10.4
pydata-google-auth            1.7.0
pydot                         1.3.0
pydot-ng                      2.0.0
pydotplus                     2.0.2
PyDrive                       1.3.1
pyemd                         0.5.1
pyerfa                        2.0.0.1
Pygments                      2.6.1
PyGObject                     3.36.0
pylev                         1.4.0
pymc                          4.1.4
PyMeeus                       0.5.12
pymongo                       4.3.3
pymystem3                     0.2.0
PyNaCl                        1.5.0
PyOpenGL                      3.1.6
pyparsing                     3.0.9
pyrsistent                    0.19.3
pysimdjson                    3.2.0
PySocks                       1.7.1
pystan                        3.3.0
pytest                        3.6.4
python-apt                    2.0.1
python-dateutil               2.8.2
python-geohash                0.8.5
python-louvain                0.16
python-slugify                8.0.0
python-utils                  3.5.0
pytz                          2022.7.1
pyviz-comms                   2.2.1
PyWavelets                    1.4.1
PyYAML                        6.0
pyzmq                         23.2.1
qdldl                         0.1.5.post3
qudida                        0.0.4
regex                         2022.6.2
requests                      2.25.1
requests-oauthlib             1.3.1
requests-unixsocket           0.2.0
resampy                       0.4.2
rpy2                          3.5.5
rsa                           4.9
scikit-image                  0.18.3
scikit-learn                  1.0.2
scipy                         1.7.3
screen-resolution-extra       0.0.0
scs                           3.2.2
seaborn                       0.11.2
Send2Trash                    1.8.0
setuptools                    57.4.0
shapely                       2.0.1
six                           1.15.0
sklearn-pandas                1.8.0
smart-open                    6.3.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soundfile                     0.11.0
spacy                         3.4.4
spacy-legacy                  3.0.12
spacy-loggers                 1.0.4
Sphinx                        3.5.4
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
SQLAlchemy                    1.4.46
sqlparse                      0.4.3
srsly                         2.4.5
statsmodels                   0.12.2
sympy                         1.7.1
tables                        3.7.0
tabulate                      0.8.10
tblib                         1.7.0
tenacity                      8.2.0
tensorboard                   2.11.2
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.1
tensorflow                    2.11.0
tensorflow-datasets           4.8.2
tensorflow-estimator          2.11.0
tensorflow-gcs-config         2.11.0
tensorflow-hub                0.12.0
tensorflow-io-gcs-filesystem  0.30.0
tensorflow-metadata           1.12.0
tensorflow-probability        0.19.0
termcolor                     2.2.0
terminado                     0.13.3
testpath                      0.6.0
text-unidecode                1.3
textblob                      0.15.3
thinc                         8.1.7
threadpoolctl                 3.1.0
tifffile                      2023.2.3
toml                          0.10.2
tomli                         2.0.1
toolz                         0.12.0
torch                         1.13.1+cu116
torchaudio                    0.13.1+cu116
torchsummary                  1.5.1
torchtext                     0.14.1
torchvision                   0.14.1+cu116
tornado                       6.0.4
tqdm                          4.64.1
traitlets                     5.7.1
tweepy                        3.10.0
typeguard                     2.7.1
typer                         0.7.0
typing_extensions             4.4.0
tzlocal                       1.5.1
uritemplate                   4.1.1
urllib3                       1.24.3
vega-datasets                 0.9.0
wasabi                        0.10.1
wcwidth                       0.2.6
webargs                       8.2.0
webencodings                  0.5.1
Werkzeug                      1.0.1
wheel                         0.38.4
widgetsnbextension            3.6.1
woodwork                      0.21.2
wordcloud                     1.8.2.2
wrapt                         1.14.1
xarray                        2022.12.0
xarray-einstats               0.5.1
xgboost                       0.90
xkit                          0.0.0
xlrd                          1.2.0
XlsxWriter                    3.0.8
xlwt                          1.3.0
yarl                          1.8.2
yellowbrick                   1.5
zict                          2.2.0
zipp                          3.12.1

**Comment by oaksharks** [comment]
This is because the `hypernets==0.2.5.1`(a project used by deeptables) do not match the latest `featurestools` since it changed API. The latest `hypernets` have fixed this problem, but the latest released `deeptables` requires `hypernets==0.2.5.1`. Try to downgrade `featuretools` again:
```
pip install featuretools==0.27
```
`featuretools==0.27` works in my local environment, good luck.
